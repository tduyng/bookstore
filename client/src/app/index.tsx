import * as React from 'react';
import { Helmet } from 'react-helmet-async';
import { Toaster } from 'react-hot-toast';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import { PATH } from './constants/paths.constant';
import { HomePage } from './pages/HomePage/Loadable';

import { NotFoundPage } from './pages/NotFoundPage/Loadable';

export function App() {
  return (
    <BrowserRouter>
      <Helmet titleTemplate="%s - Bookstore" defaultTitle="Bookstore">
        <meta name="description" content="Bookstore application" />
      </Helmet>
      <Toaster
        position="top-right"
        reverseOrder={false}
        toastOptions={{
          style: { fontSize: '1.6rem' },
        }}
      />

      <Switch>
        {/* <Route path="/books/genre/:genre" component={BooksGenre} />
        <Route path="/books/search" component={BooksSearch} />
        <Route path="/account" component={Account} />
        <Route path="/book/:id" component={BookDetail} />
        <Route path="/checkout/payment" component={Payment} />
        <Route path="/checkout" component={Checkout} exact /> */}
        <Route exact path={PATH.HOME} component={HomePage} />
        <Route component={NotFoundPage} />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
