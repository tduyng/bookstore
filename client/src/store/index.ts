import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import {
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
// import storage from 'redux-persist/lib/storage';

// import booksReducer from '../features/books/booksSlice';
// import uiReducer from '../features/ui/uiSlice';
// import userReducer from '../features/user/userSlice';

export const store = configureStore({
  reducer: {},
  middleware: getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }),
});

export const persistor = persistStore(store);
