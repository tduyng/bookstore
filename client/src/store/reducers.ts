// import booksReducer from '../features/books/booksSlice';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/es/storage';
import { bookReducer } from 'src/app/features/book/book.slice';
import { IBookReducer } from 'src/app/features/book/book.types';
import { userReducer } from 'src/app/features/user/user.slice';
import { IUserReducer } from 'src/app/features/user/user.types';

const userPersistConfig = {
  key: 'user',
  storage,
  blacklist: ['errorMsg', 'successMsg'],
};
export const rootReducers = {
  book: bookReducer,
  user: persistReducer(userPersistConfig, userReducer),
};
export interface AppState {
  user: IUserReducer;
  book: IBookReducer;
}
