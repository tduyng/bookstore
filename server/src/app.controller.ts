import { Controller, Get, HttpStatus, Response } from '@nestjs/common';
import { InjectConnection } from '@nestjs/mongoose';
import { ApiTags } from '@nestjs/swagger';
import { Connection } from 'mongoose';
import { AppService } from './app.service';

@Controller()
@ApiTags('Root')
export class AppController {
	constructor(
		private readonly appService: AppService,
		@InjectConnection() private readonly connection: Connection,
	) {}

	@Get()
	getHello(): string {
		return this.appService.getHello();
	}

	@Get('/health')
	healthCheck(@Response() res) {
		if (this.connection.readyState === 1) {
			res.status(HttpStatus.OK).json({ db: { status: 'up' } });
		} else {
			res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({ db: { status: 'down' } });
		}
	}
}
