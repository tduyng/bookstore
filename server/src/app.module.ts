import { Module } from '@nestjs/common';
import { MongooseModule } from '@nestjs/mongoose';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { envConfig } from './common/config/env.config';

@Module({
	imports: [
		MongooseModule.forRootAsync({
			useFactory: () => ({
				uri: envConfig().mongodbUri,
				useNewUrlParser: true,
				useFindAndModify: false,
				useCreateIndex: true,
			}),
		}),
	],
	controllers: [AppController],
	providers: [AppService],
})
export class AppModule {}
