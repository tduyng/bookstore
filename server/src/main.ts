import { NestFactory, Reflector } from '@nestjs/core';
import { AppModule } from './app.module';
import express from 'express';
import { NestExpressApplication } from '@nestjs/platform-express';
import { envConfig } from './common/config/env.config';
import cookieParser from 'cookie-parser';
import helmet from 'helmet';
import compression from 'compression';
import RateLimit from 'express-rate-limit';
import { setupSwagger } from './common/config/swagger.config';
import { sessionConfig } from './common/config/session.config';
import { ClassSerializerInterceptor, ValidationPipe } from '@nestjs/common';
import session from 'express-session';
import passport from 'passport';
import { useContainer } from 'class-validator';

async function bootstrap() {
	const env = envConfig();
	const app = await NestFactory.create<NestExpressApplication>(AppModule, {
		logger:
			env.mode === 'development'
				? ['log', 'debug', 'error', 'verbose', 'warn']
				: ['error', 'warn'],
	});
	const port = env.port;

	// Handle errors
	app.useGlobalPipes(
		new ValidationPipe({
			skipMissingProperties: false,
			transform: true,
			validationError: {
				target: false,
			},
		}),
	);
	app.useGlobalInterceptors(new ClassSerializerInterceptor(app.get(Reflector)));

	// Allow inject dependency injection in  validator
	useContainer(app.select(AppModule), { fallbackOnErrors: true });

	app.set('trust proxy', 1);
	app.use(express.json());
	app.use(express.urlencoded({ extended: true }));

	app.use(cookieParser(env.cookieSecret));
	app.enableCors({
		origin: true,
		methods: 'GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS',
		credentials: true,
	});

	if (env.mode === 'production') {
		app.enable('trust proxy'); // only if you're behind a reverse proxy (Heroku, Bluemix, AWS ELB, Nginx, etc)
		app
			.use(compression())
			.use(helmet())
			.use(
				RateLimit({
					windowMs: 15 * 60 * 1000, // 15 minutes
					max: 100, // limit each IP to 100 requests per windowMs
				}),
			);

		// Enable cors middleware
		app.use(function (req, res, next) {
			res.header('Access-Control-Allow-Origin', env.clientUrl); // update to match the domain you will make the request from
			res.header(
				'Access-Control-Allow-Headers',
				'Origin, X-Requested-With, X-HTTP-Method-Override, Content-Type, Accept',
			);
			res.header('Access-Control-Allow-Methods', 'GET,POST,PUT,PATCH,DELETE, OPTIONS');
			res.header('Access-Control-Allow-Credentials', true);
			if (req.method === 'OPTIONS') {
				return res.sendStatus(204);
			}
			next();
		});

		// Disable console.log() in production
		// eslint-disable-next-line @typescript-eslint/no-empty-function
		console.log = function () {};
	}

	// Session
	const sessionOptions = sessionConfig();
	app.use(session(sessionOptions));

	// Init passport
	app.use(passport.initialize());
	app.use(passport.session());

	app.setGlobalPrefix('api');
	if (env.mode !== 'production') {
		setupSwagger(app);
	}
	await app.listen(port, () => {
		console.log(`Server is running at http://localhost:${port}/api/docs/`);
	});
}
bootstrap();
