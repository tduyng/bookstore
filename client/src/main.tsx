import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import { PersistGate } from 'redux-persist/integration/react';
import { HelmetProvider } from 'react-helmet-async';
import { store, persistor } from './store';
import { Provider } from 'react-redux';
import './assets/styles/css/style.css';

ReactDOM.render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <HelmetProvider>
        <App />
      </HelmetProvider>
    </PersistGate>
  </Provider>,
  document.getElementById('root'),
);
