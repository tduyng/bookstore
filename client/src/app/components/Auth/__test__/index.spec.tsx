import React from 'react';
import { Provider } from 'react-redux';
import { store } from 'src/store';
import { Auth } from '..';
import { createRenderer } from 'react-test-renderer/shallow';

const renderer = createRenderer();

describe('<Auth/>', () => {
  it('Should match snapshot', () => {
    renderer.render(
      <Provider store={store}>
        <Auth open={true} onClose={() => jest.fn()} />
      </Provider>,
    );
    const renderOutput = renderer.getRenderOutput();

    expect(renderOutput).toMatchSnapshot();
  });
});
