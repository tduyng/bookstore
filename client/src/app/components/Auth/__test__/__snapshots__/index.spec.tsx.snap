// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth/> Should match snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Auth
    onClose={[Function]}
    open={true}
  />
</Context.Provider>
`;
